xc/device.ml

```diff
@@ -66,6 +66,9 @@ module Socket = struct
   type t = Unix of string | Port of int
   module Unix = struct
     let path x = "unix:" ^ x
+    let rm   x = try (* best effort *)
+      if Sys.file_exists x then Sys.remove x
+      with e -> debug "error cleaning unix socket %s: %s" x (Printexc.to_string e)
```

Comment:
I was going to suggest to simply do `Sys.remove` and ignore the specific 'No such file or directory' issue to avoid a race here, however the errors in the `Sys` module are horrible (it raises `Sys_error string`) and impossible to pattern match reliably, so this is fine as it is.

I think it would be an improvement to avoid using `Sys` and instead do:
```ocaml
try
Unix.unlink x
with
| Unix.Unix_error (Unix.ENOENT, "unlink", _) -> ()
| e -> debug "error cleaning unix socket %s: %s" x (Unix.error_message e)
```

Reply:
...but that's again matching on a string inside the the `Unix_error`.

Reply:
Well, that's the name of the command, and we are running just that one. But in fact it would not be needed, we just `unlink` so we might as well simply do
```ocaml
try Unix.unlink x
with
| Unix.Unix_error (Unix.ENOENT, _, _) -> ()
| e -> debug "error cleaning unix socket %s: %s" x (Unix.error_message e)
```

Reply:
Exactly. You'll still get the compiler warning otherwise.